keras/layers/convolutional.py

```diff
@@ -3400,6 +3400,12 @@ def compute_output_shape(self, input_shape):
     return tf.TensorShape([input_shape[0], length, input_shape[2]])
 
   def call(self, inputs):
+    if sum(self.cropping) >= inputs.shape[1]:
```

Comment:
`inputs.shape[1]` can be None. The check would need to use `tf.shape(inputs)`. In fact you probably want to use https://www.tensorflow.org/api_docs/python/tf/debugging/assert_greater

Also: why `sum`?

Reply:
If the inputs of the layer can be an array of `None`  yes we have to discard this situation.

`sum` because self.cropping` is a tuple of two numbers integers, the first integer is the number of rows to cut from the top, and the second integer is the number of rows to cut from the bottom. If the sum of this tuple is higher than the number of rows (`inputs.shape[1]`) them the result will be an empty list (wich we want to avoid).

Reply:
The input itself should not be None, but the static shape could be None on multiple dimensions. Static shape = the shape accessed with tensor.shape, see the [shape docs](https://www.tensorflow.org/api_docs/python/tf/shape).

This would come up when tracing a [tf.function](https://www.tensorflow.org/api_docs/python/tf/function). This python code would only be run once, on inputs of unknown shape, and the compiled graph run on tensors of potentially multiple different shapes. Using tf.debugging.assert_greater would allow this error to keep working when the graph is compiled.

Alternately, we could only log this error when the static shape is known (not None). That could be OK, as if you are really calling this layer on multiple shaped inputs, you might not want a single small input to take down your training run. In that case this if case could stay largely the same, but you do still need to guard against None in `inputs.shape`.